### PR TITLE
LPS-173628 Add permission check on account roles

### PR DESCRIPTION
**Pull request by smottal**
https://liferay.atlassian.net/browse/LPS-173628

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:um

File: modules/apps/account/account-service/src/main/java/com/liferay/account/internal/security/permission/resource/AccountRoleModelResourcePermission.java [hunk 1]
```diff
@@ -168,6 +178,63 @@ public PortletResourcePermission getPortletResourcePermission() {
...
+
+			Organization organization =
+				_organizationLocalService.fetchOrganization(
+					accountEntryOrganizationRel.getOrganizationId());
+
+			Organization originalOrganization = organization;
+
+			while (organization != null) {
+				boolean organizationMember = ArrayUtil.contains(
+					userOrganizationIds, organization.getOrganizationId());
```
Comment (andrea-ale-sbarra) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/internal/security/permission/resource/AccountRoleModelResourcePermission.java:209]: can you move this closer to the "if" where it is used?

File: modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java [hunk 2]
```diff
@@ -274,35 +276,49 @@ public BaseModelSearchResult<AccountRole> searchAccountRoles(
...
+		}
 
-		return new BaseModelSearchResult<AccountRole>(
-			TransformUtil.transform(
-				searchHits.getSearchHits(),
-				searchHit -> {
-					Document document = searchHit.getDocument();
+		try (SafeCloseable safeCloseable =
+				AccountRolePermissionThreadLocal.setWithSafeCloseable(
+					accountEntryId)) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java:288]: why do you need this?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java (hunk 2 above)
Comment (smottal) [reply modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountRoleLocalServiceImpl.java:254]: To make available the accountEntryId in the PermissionChecker

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:um

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:um

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Review by luca-pellizzon — APPROVED** [review]

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
New PR: https://github.com/liferay-commerce/liferay-portal/pull/4331

**Pull request closed without merging**